src/JsonSchema/Constraints/ConstraintInterface.php

```diff
@@ -59,5 +59,10 @@ public function isValid();
      * @param mixed            $i
      * @throws \JsonSchema\Exception\ExceptionInterface
      */
+    public function _check(&$value, $schema = null, JsonPointer $path = null, $i = null);
```

Comment:
My original suggestion from last night was to bite the bullet and modify `check` (which is already public) to take a reference. Yes, it will break bc for some people, but I think we have enough rationale at this point to justify making the change. If you want to do it in steps, we could begin by replacing `coerce` with `validate` (which takes a reference, but otherwise makes no assumptions about what you want to do, and bringing the grand total of public methods to 3), and deprecating `check`, then eventually removing `check`.